contrib/cirrus/lib.sh.t

```diff
@@ -83,6 +83,46 @@ test_rev "FOO BAR" 9 'FATAL: test_rev() requires $BAR to be non-empty'
 BAR=1
 test_rev "FOO BAR" 0 ''
 
+###############################################################################
+# tests for test_okay()
+
+function test_test_okay {
+    local act=$1
+    local op=$2
+    local exp=$3
+    local desc=$4
+    local expected_status=$5
+    local expected_msg="$6"
+    local msg
+    if [[ -n "$act" ]]; then
+        msg=$(test_okay "$act" "$op" "$exp" "$desc")
+        local status=$?
+    else  # 3-argument form
+        msg=$(test_okay "$op" "$exp" "$desc")
+        local status=$?
+    fi
+
+    check_result "$msg"    "$expected_msg"    "test_okay $act $op $exp $desc"
+    check_result "$status" "$expected_status" "test_okay $act $op $exp $desc (actual rc $status)"
+}
+
+# negative tests
+# Be careful -  test_okay() supports 3 or 4 arguments
+test_test_okay "" "" "" "" 9 'FATAL: test_okay() requires $OP to be non-empty'
+test_test_okay "foo" "" "" "" 9 'FATAL: test_okay() requires $OP to be non-empty'
+test_test_okay "foo" "bar" "" "" 9 'FATAL: test_okay() requires $RHS to be non-empty'
+test_test_okay "foo" "bar" "baz" "" 9 'FATAL: test_okay() requires $DESC to be non-empty'
+test_test_okay "foo" "bar" "baz" "fail" 2 'not ok - fail: foo bar baz: bar: binary operator expected'
+test_test_okay "foo" "-gt" "baz" "FAIL" 2 'not ok - FAIL: foo -gt baz: foo: integer expression expected'
+test_test_okay "1" "-ne" "0" 'Basic Logic' 0 'ok - Basic Logic: 1 -ne 0'
+test_test_okay "some string" "==" "other string" 'Hmmm' 1 'not ok - Hmmm: some string == other string'
+
+# positive tests
+test_test_okay "1" "-ne" "0" 'logic' 0 'ok - logic: 1 -ne 0'
+test_test_okay "" "-r" "$0" 'Darth Vader'"'"'s Credit Card Number' 0 'ok - Darth Vader'"'"'s Credit Card Number: -r contrib/cirrus/lib.sh.t'
```

Comment:
This doesn't work if the test is invoked as `./contrib/etc`, or `./lib.sh.t` from the same directory, etc. (And, the test name is completely weird).